src/httpfetch.cpp

```
@@ -275,55 +275,60 @@ HTTPFetchOngoing::HTTPFetchOngoing(const HTTPFetchRequest &request_,
 		curl_easy_setopt(curl, CURLOPT_WRITEDATA, &result.data);
 	}
 
+	// Configure the method
+	switch (request.method) {
+	case HTTP_GET:
+		curl_easy_setopt(curl, CURLOPT_HTTPGET, 1);
+		break;
+	case HTTP_POST:
+		curl_easy_setopt(curl, CURLOPT_POST, 1);
+		break;
+	case HTTP_PUT:
+		curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "PUT");
+		break;
+	case HTTP_DELETE:
+		curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "DELETE");
+		break;
+	}
+
```

Else? Fail silently? Perhaps log a warning to uncover such incorrect API uses.